Index: luv-1.9.0-3/src/handle.c
===================================================================
--- luv-1.9.0-3.orig/src/handle.c	2016-05-17 11:50:25.000000000 +0200
+++ luv-1.9.0-3/src/handle.c	2016-05-22 12:06:56.238738783 +0200
@@ -30,8 +30,10 @@
 
 static uv_handle_t* luv_check_handle(lua_State* L, int index) {
   int isHandle;
-  uv_handle_t* handle;
-  if (!(handle = *(void**)lua_touserdata(L, index))) { goto fail; }
+  uv_handle_t** phandle = (uv_handle_t **)lua_touserdata(L, index);
+  if (!phandle) { goto fail; }
+  uv_handle_t* handle = *phandle;
+  if (!handle) { goto fail; }
   lua_getfield(L, LUA_REGISTRYINDEX, "uv_handle");
   lua_getmetatable(L, index < 0 ? index - 1 : index);
   lua_rawget(L, -2);
