Index: luvi-0.1/cmake/Modules/LuaJITAddExecutable.cmake
===================================================================
--- luvi-0.1.orig/cmake/Modules/LuaJITAddExecutable.cmake	2015-05-28 15:17:13.420320674 +0200
+++ luvi-0.1/cmake/Modules/LuaJITAddExecutable.cmake	2015-05-28 15:17:37.452319716 +0200
@@ -8,7 +8,8 @@
 # The Luvit Authors
 
 MACRO(LUAJIT_add_custom_commands luajit_target)
-  SET(TARGET_ARCH $ENV{TARGET_ARCH})
+  SET(TARGET_ARCH arm)
+  MESSAGE(STATUS ".................... LuaJIT ARCH=${TARGET_ARCH}")
   SET(target_srcs "")
   FOREACH(file ${ARGN})
     IF(${file} MATCHES ".*\\.lua$")
