Index: luvi-2.7.2/deps/lua-openssl/src/ssl.c
===================================================================
--- luvi-2.7.2.orig/deps/lua-openssl/src/ssl.c	2016-08-08 11:52:48.000000000 +0200
+++ luvi-2.7.2/deps/lua-openssl/src/ssl.c	2016-08-13 14:02:50.202352739 +0200
@@ -717,6 +717,14 @@
   return ec_tmp;
 }
 
+static int openssl_ssl_ctx_set_ecdh_auto(lua_State *L)
+{
+  SSL_CTX* ctx = CHECK_OBJECT(1, SSL_CTX, "openssl.ssl_ctx");
+  int on_off = lua_toboolean(L, 2);
+  SSL_CTX_set_ecdh_auto(ctx, on_off);
+  return 0;
+}
+
 static int openssl_ssl_ctx_set_tmp(lua_State *L)
 {
   SSL_CTX* ctx = CHECK_OBJECT(1, SSL_CTX, "openssl.ssl_ctx");
@@ -996,6 +1004,7 @@
 
   {"verify_depth",    openssl_ssl_ctx_verify_depth},
   {"set_tmp",         openssl_ssl_ctx_set_tmp},
+  {"set_ecdh_auto",   openssl_ssl_ctx_set_ecdh_auto},
   {"flush_sessions",  openssl_ssl_ctx_flush_sessions},
   {"session",         openssl_ssl_ctx_sessions},
   {"session_cache_mode",        openssl_session_cache_mode },
@@ -1297,6 +1306,7 @@
 static int openssl_ssl_pushresult(lua_State* L, SSL*ssl, int ret_code)
 {
   int err = SSL_get_error(ssl, ret_code);
+
   switch (err)
   {
   case SSL_ERROR_NONE:
